tsl/src/compression/create.c

```diff
@@ -674,6 +674,23 @@ add_time_to_order_by_if_not_included(List *orderby_cols, List *segmentby_cols, H
 	return orderby_cols;
 }
 
+static FormData_hypertable_compression *
+get_col_info_for_attnum(Hypertable *ht, CompressColInfo *colinfo, AttrNumber attno)
+{
+	Relation rel = relation_open(ht->main_table_relid, AccessShareLock);
+	Form_pg_attribute attr = TupleDescAttr(RelationGetDescr(rel), AttrNumberGetAttrOffset(attno));
+	int colno;
+	FormData_hypertable_compression *res = NULL;
+
+	for (colno = 0; colno < colinfo->numcols; colno++)
+	{
+		if (namestrcmp(&colinfo->col_meta[colno].attname, NameStr(attr->attname)) == 0)
```

Comment:
Shouldn't there be a check for `attisdropped` here or at least an Assert?

Reply:
changed to using get_attname here so don't have that info, but double checked and that can't happen.